tests/test_apply.py

```diff
@@ -51,11 +51,46 @@
     qml.T
 ]
 
+test_transpile_options = [
+    {},
+    {'optimization_level':2},
+    {'optimization_level':2, 'seed_transpiler':22}
+]
+
+test_device_options = [
+    {},
+    {'optimization_level':3},
+    {'optimization_level':1}
+]
+
 single_qubit_operations_param = [qml.PhaseShift, qml.RX, qml.RY, qml.RZ]
 two_qubit = [qml.CNOT, qml.SWAP, qml.CZ]
 two_qubit_param = [qml.CRZ]
 three_qubit = [qml.Toffoli, qml.CSWAP]
 
+@pytest.mark.parametrize("qiskit_device", [AerDevice, BasicAerDevice])
+@pytest.mark.parametrize("wires", [1,2])
+@pytest.mark.parametrize("transpile_options", test_transpile_options)
+@pytest.mark.parametrize("device_options", test_device_options)
+@pytest.mark.parametrize("persist", [True, False])
+@pytest.mark.transpile_args_test
+class TestTranspileOptionsUpdations:
+    """Test the updation of transpilation options passed while applying operations"""
```

Comment:
```suggestion
class TestTranspileOptionsUpdate:
    """Test the update of transpilation options passed while applying operations"""
```

Reply:
As per new implementation, this file won't have these test cases.